core/xml.js

```diff
@@ -332,6 +338,71 @@ Blockly.Xml.domToWorkspace = function(xml, workspace) {
 };
 
 /**
+ * Decode an XML DOM and create blocks on the workspace.
+ * Append the new blocks at the bottom of the existing blocks
+ * @param {!Element} xml XML DOM.
+ * @param {!Blockly.Workspace} workspace The workspace.
+ */
+Blockly.Xml.appendDomToWorkspace = function(xml, workspace) {
+  
+  // find bottom position and alignment of the current blocks  
+  //var blocks = workspace.getAllBlocks();
+  var blocks = workspace.getTopBlocks();
+  var offsetY = 0; // container for the y of the bottom block
+  var offsetX = 0; 
+  var blocksCount = blocks.length;
+  if (blocksCount > 0) {
+	// initialize the position of the corners using position 
+	// of the first block
+	var bHW = (("height" in blocks[0])? blocks[0].getHeightWidth() : {height: 0, width: 0});
```

Comment:
JavaScript is a single threaded, so is mostly safe.  Do this in a try finally block, to protect against exceptions:

```
var savetab = Blockly.BlockSvg.TAB_WIDTH;
try {
  Blockly.BlockSvg.TAB_WIDTH = 0;
  var bbox = workspace.getBlocksBoundingBox();
} finally {
  Blockly.BlockSvg.TAB_WIDTH = savetab;
}
```